README.md

```diff
@@ -306,49 +649,92 @@ dojo --docker-options="--init" --image=alpine:3.8 -i=false sh -c "echo 'will sle
+```
+vagrant up
+vagrant ssh
+vagrant down
 ```
+Major difference is that:
+ * vagrant assumes a long running development environment, rather then spinning up new host for running each command. Dojo allows to have both approaches.
+ * `vagrant up` can take time (VM booting time)
+ * vagrant (in most cases) requires a periodic synchronization of your current working directory to the VM and back. In dojo, the filesystem is shared between the container and the host.
 
-Alternatively, you can build the binary file yourself, see [Development](#development).
+Still vagrant is a great tool and irreplaceable if you must use a VM.
 
-## Aims
-Dojo is intended to be used with **short-running commands**, e.g. to compile code, to create backup dump, to use a 3rd party software to do some operation like e.g. use [Terraform](https://www.terraform.io/) to create a VM or run some graphical IDE for local development. It is rather not expected to use Dojo with long-running commands, e.g. to run a server 24/7.
+`Dojofile` is similar to `Vagrantfile` in its function. But with a restriction that Dojo forces you to first build and release the image, while `Vagrantfile` has some tools to provision the VM when running `vagrant up` which adds up to the startup time.
 
-Dojo was invented in order to resolve the problems of:
-   * **fast changing work environment requirements** (you had to reinstall/reconfigure over and over again)
-   * using **different environments at the same moment** (e.g using Java 7 for some projects and Java 8 for other)
-   * **work environments becoming bloated** (in one machine you have to have installed and configured: Mono, Java, Python, etc.)
-   * **not reproducible work environments** - **works on my machine** (e.g. you installed Mono just fine on your work computer, but then installing it on your laptop demanded some additional steps or resulted in having installed different versions). This is especially important if you use  **Continuous Integration**, the environment should be the same on workstation and CI agents.
-   * **mutable work environments** - e.g. your code compiled a month ago, but you installed sth new and it influenced the environment
-   * **experimenting with new technologies without breaking your workstation**
+### vs. dependency managers
 
-## Limitations
-Dojo was tested only on Linux, with local Docker server.
+A typical dependency manager allows you to commit references to all immediate and transitive dependencies in source control, files such as `Gemfile.lock`, `paket.lock`, `yarn.lock` etc. It is a good practice to do so, because project defines fully, in code, what libraries it depends on.
+If we extend this concept, then we soon realize that the project did not define how the host, where it should be built, should look like. Probably most common situation is that there is a README file listing several tools to install. It is also the least helpful situation.
```

Comment:
> It is also the least helpful situation.

~~This is subjective. Let's change it to: "Such an environment specification is not enough."~~ I've read the next paragraphs and understood that sentence. Maybe let's change it to something like: "It is also the least helpful situation among the situations described in this section".